File: 8/list-8.py
class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregado = []

    def inserir_dados_empregado(self, nome, cargo, salario):
        self.empregado.append(Empregado(nome, cargo, salario))

    def listar_empresa(self):
        for empregado in self.empregado:
            print(empregado.nome, empregado.cargo, empregado.salario, self.nome)

class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

def main():
    empresa = Empresa("Dataprev")
    
    empresa.inserir_dados_empregado("Marcio", "IT", 10000)
    empresa.inserir_dados_empregado("Roberto", "Programador", 40000)

    empresa.listar_empresa()

if __name__ == "__main__":
    main()

File: 8/list-8.go
package main

import "fmt"

type Empresa struct {
	nome      string
	empregado []*Empregado
}

func (e *Empresa) adicionarEmpregado(em *Empregado) {
	e.empregado = append(e.empregado, em)
}

type Empregado struct {
	nome    string
	cargo   string
	salario float64
}

func main() {

	adcionar := &Empregado{nome: "Marcio", cargo: "IT", salario: 12000}
	Empresa := &Empresa{nome: "Dataprev"}

	Empresa.adicionarEmpregado(adcionar)

	fmt.Printf("Empresa: %s tem empregados: ", Empresa.nome)
	for _, em := range Empresa.empregado {
		fmt.Printf("%s com cargo %s e salario %.2f\n", em.nome, em.cargo, em.salario)
	}

}
